pkg/controller/migcluster/validation.go

```diff
@@ -0,0 +1,40 @@
+package migcluster
+
+import (
+	"context"
+	migapi "github.com/fusor/mig-controller/pkg/apis/migration/v1alpha1"
+)
+
+// Types
+const (
+	Ready = "Ready"
+)
+
+// Reasons
+const ()
+
+// Statuses
+const (
+	True  = "True"
+	False = "False"
+)
+
+// Messages
+const (
+	ReadyMessage = "The cluster is ready."
+)
+
+// Validate the asset collection resource.
+// Returns error and the total error conditions set.
+func (r ReconcileMigCluster) validate(assetCollection *migapi.MigCluster) (error, int) {
+	totalSet := 0
+	var err error
+
+	// Apply changes
+	err = r.Update(context.TODO(), assetCollection)
```

Comment:
Curious about these... If the `r.Update()` fails because the resource has been modified by someone else, do we requeue the item so that we can attempt to run the validation again later?

Reply:
The error should be passed all the way up to the reconcile main loop and result in a failed reconcile.  It's my understanding that this results in a requeue.